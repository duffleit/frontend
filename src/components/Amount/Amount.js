import React from "react";
import { FormattedNumber } from "react-intl";
import { connect } from "react-redux";
import { getCurrency } from "../../routes/App/modules/currentNobt/selectors";

const valueViewModel = (props) => (props.absolute) ? Math.abs(props.value): props.value;
const computeValueClass = (props) => (props.value > 0) ? props.theme.positive : props.theme.negative

const Amount = (props) => (
  <span className={props.theme.root + " " + computeValueClass(props)}>
    <FormattedNumber value={valueViewModel(props)} currency={props.currency} style="currency"/>
  </span>
);

Amount.defaultProps = {
  absolute: true,
  theme: {
    root: "",
    positive: "",
    negative: ""
  }
};

Amount.propTypes = {
  value: React.PropTypes.number.isRequired,
  absolute: React.PropTypes.bool,
  theme: React.PropTypes.shape({
    root: React.PropTypes.string,
    positive: React.PropTypes.string,
    negative: React.PropTypes.string
  })
};

export default connect(
  (state) => ({
    currency: getCurrency(state)
  }),
  () => ({})
)(Amount)
