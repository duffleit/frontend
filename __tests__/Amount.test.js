import React from "react";
import renderer from "react-test-renderer";
import Amount from "components/Amount";
import CurrencyContext from "../__mocks__/currencyContext";


test('should use absolute value of amount by dafult', () => {

  var cmp = renderer.create(
    <CurrencyContext currency="EUR">
      <Amount value={-30} />
    </CurrencyContext>
  );

  var tree = cmp.toJSON();

  expect(tree).toMatchSnapshot();
});

test('should display actual value if absolute property is falsy', () => {

  var cmp = renderer.create(
    <CurrencyContext currency="EUR">
      <Amount value={-30} absolute={false} />
    </CurrencyContext>
  );

  var tree = cmp.toJSON();

  expect(tree).toMatchSnapshot();
});
