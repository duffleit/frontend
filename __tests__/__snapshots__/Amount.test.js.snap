exports[`test should display actual value if absolute property is falsy 1`] = `
<span
  className="">
  <span>
    -30
  </span>
</span>
`;

exports[`test should use absolute value of amount by dafult 1`] = `
<span
  className="">
  <span>
    30
  </span>
</span>
`;
